### PR TITLE
fix: signatures for tests

### PR DESCRIPTION
**Pull request by zheeeng**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [x] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [x] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | Update the types in test cases |
| 🇨🇳 Chinese | 更新测试用例中的类型 |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/_util/wave.tsx [hunk 1]
```diff
@@ -237,8 +237,4 @@ class InternalWave extends React.Component<WaveProps> {
   }
 }
 
-const Wave = forwardRef<InternalWave, WaveProps>((props, ref) => (
```
Comment (zheeeng) [inline components/_util/wave.tsx:240]: class 组件不需要包裹 forwardRef

File: components/checkbox/__tests__/group.test.tsx [hunk 2]
```diff
@@ -242,10 +242,10 @@ describe('CheckboxGroup', () => {
     const onChange = jest.fn();
 
     const Demo: React.FC = () => {
-      const [v, setV] = useState<string>('');
+      const [v, setV] = useState('');
 
       React.useEffect(() => {
-        setTimeout(setV('1') as unknown as TimerHandler, 1000);
```
Comment (zheeeng) [inline components/checkbox/__tests__/group.test.tsx:248]: 这里之前的写法不知道用意是什么。等价于
setV('1')

setTimeout(undefined, 1000)

Comment (li-jia-nan) [reply components/checkbox/__tests__/group.test.tsx:248]: 这个文件好像是我重构过的，之前是js，用ts重构了一遍

当时我也注意到这里了，不太懂他的用意，所以只是加了类型，解决了一下 type 报错

Comment (zheeeng) [reply components/checkbox/__tests__/group.test.tsx:248]: 所以也没敢动，这个测试用例没看太懂

Comment (li-jia-nan) [reply components/checkbox/__tests__/group.test.tsx:248]: @zombieJ @MadCcc

File: .eslintrc.js [hunk 3]
```diff
@@ -31,7 +31,7 @@ module.exports = {
         '@typescript-eslint/no-unused-vars': [2, { args: 'none' }],
         'no-unused-expressions': 'off',
         '@typescript-eslint/no-unused-expressions': 2,
-        '@typescript-eslint/consistent-type-imports': 2,
+        '@typescript-eslint/consistent-type-imports': [2, { disallowTypeAnnotations: false }],
```
Comment (afc163) [inline .eslintrc.js:34]: 原因是？

Comment (zheeeng) [reply .eslintrc.js:34]: 支持这种用法 const a: import('somemodule') = whatever

一般 `import type {} from 'somemodule'` 挺好的，它不会引入模块的副作用，但是没有 `import type xxx from 'somemodule'` 把默认导出当类型的用法。虽然可以 `import xxx from 'somemodule‘; type XXX = typeof xxx`，但是会触发模块的副作用。遇到有默认导出的情况，只好用 `const a: typeof import('somemodule').default = whatever`。得修改 eslint

如果可能，大家都不再用默认导出就好了。

**Review by li-jia-nan — APPROVED** [review]

**Pull request merged**